File: word_break.cpp
class Solution { // 24ms
public:
    bool wordBreak(string s, vector<string>& wordDict) {
        vector<bool> dp(s.size(),false);
        dp[0] = true;
        for(int i = 1; i<s.size()+1; i++){
            for(int j=0; j<i; j++){
                // cout<<s.substr(j,i-j)<<endl;
                if(dp[j] && find(wordDict.begin(), wordDict.end(), s.substr(j,i-j)) != wordDict.end() ){
                    dp[i] = true;
                    break;
                }
            }
        }
        return dp[s.size()];
    }
};

File: word_break.py
class Solution: # 44ms
    def wordBreak(self, s: 'str', wordDict: 'List[str]') -> 'bool':
        dp = [False for i in range(len(s)+1)]
        wordSet = wordDict
        dp[0] = True
        for i in range(1, len(s)+1):
            # print(dp)
            for j in range(i):
                # print(s[j:i])
                if dp[j] and s[j:i] in wordSet:
                    dp[i] = True
                    break
        return dp[len(s)]